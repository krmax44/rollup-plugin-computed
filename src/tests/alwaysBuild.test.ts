import type { Computers } from '..';
import { roll } from './utils';

describe('alwaysBuild', () => {
	it('force builds computes', async () => {
		expect.assertions(2);

		const computers: Computers = {
			test: {
				alwaysBuild: true,
				type: 'asset',
				fileExt: 'json',
				fn() {
					return JSON.stringify({ it: 'works' });
				}
			}
		};

		const output = await roll('nothing.js', computers);
		const file: any = output.find(c => c.fileName === 'assets/test.json');

		expect(output.length).toBe(2);
		expect(file.source).toMatchSnapshot();
	});
});
