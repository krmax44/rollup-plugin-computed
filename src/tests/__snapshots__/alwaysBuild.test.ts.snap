// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alwaysBuild force builds computes 1`] = `"{\\"it\\":\\"works\\"}"`;
